iree/turbine/kernel/wave/decompose_reduce_ops.py

```diff
@@ -67,11 +65,11 @@ def determine_shuffle_config(
     offset = access_pattern.start.subs({k: 0 for k in ignore})
     offset = subs_idxc(offset)
     offset_table = [offset.subs({THREAD_0: i}) for i in range(subgroup_size)]
-    # Determine the thread ids participating in the shuffle.
+    unique_offsets = list(dict.fromkeys(offset_table))
```

Comment:
I don't quite understand this, isn't offset_table of type `list[IndexExpr]`?

Reply:
Ah, we're just trying to get a set of the "unique" offsets

Reply:
Good question. Yes you are correct that we do not use set so that we can maintain the ordering. But the ordering is not important for expansion or figuring out the indexing dimensions. In this case maintaining the ordering is important so that when we compute the cluster stride we get the correct value. For example, say we have the following unique threads [0, 4, 8, 12]. If we do not maintain the ordering, then we could end up with the following set [0, 8, 4, 12] and now when we do a diff to compute the cluster stride we would get [8, -4, 8] as opposed to [4, 4, 4] which is what we really want.

Reply:
makes sense, thanks!